src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -112,12 +119,16 @@ public void setPerson(Person target, Person editedPerson) {
      */
     public void removePerson(Person key) {
         persons.remove(key);
-        tasks.remove(key.getTask());
+        Task task = key.getTask();
+        if (task != null) {
```

Comment:
Good to handle null pointer exception when attempting to remove the task associated with the person.